### PR TITLE
Make GPU-aware MPI a runtime option

### PR DESCRIPTION
**Pull request by liruipeng**
This PR makes GPU-aware MPI a runtime option for further testing and development. 

- [x] tux
- [x] lassen
- [x] tioga

### CODE REVIEW

**Review by rfalgout — APPROVED** [review]
Looks good, Rui Peng!

File: src/utilities/handle.c [hunk 1]
```diff
@@ -182,3 +182,22 @@ hypre_SetUserDeviceMfree(GPUMfreeFunc func)
 
    return hypre_error_flag;
 }
+
+HYPRE_Int
+hypre_SetGpuAwareMPI( HYPRE_Int use_gpu_aware_mpi )
+{
+#if defined(HYPRE_USING_GPU) || defined(HYPRE_USING_DEVICE_OPENMP)
+   hypre_HandleUseGpuAwareMPI(hypre_handle()) = use_gpu_aware_mpi;
```
Comment (victorapm) [inline src/utilities/handle.c:190]:
```suggestion
   hypre_HandleUseGpuAwareMPI(hypre_handle()) = use_gpu_aware_mpi;
#else
   HYPRE_UNUSED_VAR(use_gpu_aware_mpi);
```

**Review by victorapm — APPROVED** [review]
Thanks!!

**Review by ulrikeyang — APPROVED** [review]
Looks good. Thank you.

**Comment by PaulMullowney** [comment]
Is building with GPU-aware MPI default now? What's the run time parameter to toggle?

**Comment by victorapm** [comment]
The default remains unchanged (no GPU-aware since hypre is configured without the option `--enable-gpu-aware-mpi`).

The new API for turning on/off GPU-aware is the following: 

https://github.com/hypre-space/hypre/blob/edfda7c8bc679f84749fd9a2ca6b70ee60b4ef37/src/test/ij.c#L2726

It can be turned on, and thus activate GPU-aware MPI, without having to recompile hypre with `--enable-gpu-aware-mpi`. However, the user still has to make sure that the MPI dependencies are linked to the executable

**Comment by PaulMullowney** [comment]
If I build with --enable-gpu-aware-mpi, do i still need to set that parameter?

**Comment by victorapm** [comment]
No

**Pull request merged**